src/app/pages/BridgeDepositPage/components/AmountSelector/index.tsx

```diff
@@ -0,0 +1,220 @@
+import React, { useCallback, useMemo, useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { bignumber } from 'mathjs';
+import styled from 'styled-components/macro';
+import type { Chain } from 'types';
+
+import { actions } from '../../slice';
+import { selectBridgeDepositPage } from '../../selectors';
+import { BridgeDictionary } from '../../dictionaries/bridge-dictionary';
+import { CrossBridgeAsset } from '../../types/cross-bridge-asset';
+import { useTokenBalance } from '../../hooks/useTokenBalance';
+import { AssetModel } from '../../types/asset-model';
+import { AmountInput } from '../AmountInput';
+import { FormGroup } from '../../../../components/Form/FormGroup';
+import { useBridgeLimits } from '../../hooks/useBridgeLimits';
+import { toNumberFormat } from '../../../../../utils/display-text/format';
+import { fromWei } from 'utils/blockchain/math-helpers';
+import { useBridgeTokenBalance } from '../../hooks/useBridgeTokenBalance';
+import { LoadableValue } from 'app/components/LoadableValue';
+import { ActionButton } from 'app/components/Form/ActionButton';
+import { translations } from 'locales/i18n';
+import { useTranslation } from 'react-i18next';
+
+export function AmountSelector() {
+  const { amount, chain, targetChain, sourceAsset, targetAsset } = useSelector(
+    selectBridgeDepositPage,
+  );
+  const { t } = useTranslation();
+  const dispatch = useDispatch();
+
+  const asset = useMemo(
+    () =>
+      BridgeDictionary.get(chain as Chain, targetChain)?.getAsset(
```

Comment:
I don't like this casting at all. As always, you could solve it with `Non-null assertion operator` but I think we have a conceptual problem here - I don't get why is almost everything `Nullable` when you cannot send `null` to the methods you need to use (e.g. `BridgeDictionary.get`, `getAsset` etc.).

I think you should rather adjust the type definitions and have `Nullable` only on the relevant places but if you find that this would lead to some massive redesign, just use `!`.